src/main/java/seedu/duke/ui/AppUi.java

```diff
@@ -198,18 +201,22 @@ public boolean matchCommand(String command, HashMap<String, String> commandArgum
 
         case "expense":
             addExpense(commandArguments);
+            this.mainStorage.update(financialList);
```

Comment:
Perhaps you could move this line to outside the case block so that you don't need to repeat this line of code for each case?